### ISSUE TITLE
Support for disable or soft delete

### ISSUE THREAD

**Issue by clintfoster** [aws/amazon-cognito-identity-js#484]
Have you considered supporting `disable()` from the cognito-identity SDK (not just the admin API)? An app would then be able to effectively "soft delete" an account in a way that the user could later reinstate it by calling customer support, who could re-enable it via the console.

If the **DISABLED** status isn't meant to be used in this way, perhaps a **DELETED** status and corresponding `softDelete()` function could be introduced.

**Comment by behrooziAWS** [comment]
Can you give an example of why an end user would want to disable their account?

**Comment by clintfoster** [comment]
Suppose a user closes their account and later has a question. We won't be able to look them up if their contact details have been deleted. Having said that, we must ensure the user retains full control of their personal information. This could be addressed with an interaction like the following:

1. User selects **Close account**.
1. "Are you sure?" dialog is presented with a checkbox:
>☐ Delete contact details. (We will not be able to look you up if you have questions later.)

If the checkbox is selected, the app calls `delete()`. Otherwise it calls `disable()` or `softDelete()`.

**Comment by behrooziAWS** [comment]
Makes sense, we will consider exposing it to the end user as a feature request.
